src/dune_pkg/dune

```diff
@@ -4,6 +4,7 @@
  (libraries
   stdune
   fiber
+  memo
```

Comment:
I would stick to fiber everywhere. Makes the API more general. Memo is only for generating the rules.

Reply:
But doesn't Memo have memorizing semantics? That seems to be useful as the result will never change across runs.

Reply:
Right, but that's for the concrete rules to decide. Also note that turning a fiber into a memo using `Memo.of_reproducible_fiber` isn't enough to give these semantics. You need to make sure the fiber isn't recomputed every time we bind on it by creating a cell.

Reply:
So if I understand you correctly we'd determine the value in the rules and would then have them as a `Memo.t` at that point?

I've adjusted the code to use fibers.

Reply:
Basically. In general, we push the rule creation down to dune_rules and package management is no exception. There should be a core library that's more widely usable and doesn't use memoization. The rules can finally use memoization to setup the rules, but other parts of the code (e.g. the command line), can use the api differently should they choose to.